packages/web-pkg/src/components/Editor.vue

```diff
@@ -0,0 +1,105 @@
+import { computed, defineComponent, unref, PropType, ref, onMounted } from 'vue'
+import { Resource } from '@ownclouders/web-client/src/helpers/resource/types'
+
+import '@toast-ui/editor/dist/toastui-editor.css'
+import '@toast-ui/editor/dist/theme/toastui-editor-dark.css'
+import '@toast-ui/editor/dist/i18n/cs-cz'
+import '@toast-ui/editor/dist/i18n/de-de'
+import '@toast-ui/editor/dist/i18n/es-es'
+import '@toast-ui/editor/dist/i18n/fr-fr'
+import '@toast-ui/editor/dist/i18n/it-it'
+import '@toast-ui/editor/dist/i18n/nl-nl'
+import '@toast-ui/editor/dist/i18n/ko-kr'
+import '@toast-ui/editor/dist/i18n/sv-se'
+import '@toast-ui/editor/dist/i18n/tr-tr'
+import '@toast-ui/editor/dist/i18n/zh-cn'
+import { Editor, EditorOptions } from '@toast-ui/editor'
+
+import 'prismjs/themes/prism.css'
+import '@toast-ui/editor-plugin-code-syntax-highlight/dist/toastui-editor-plugin-code-syntax-highlight.css'
+import codeSyntaxHighlight from '@toast-ui/editor-plugin-code-syntax-highlight/dist/toastui-editor-plugin-code-syntax-highlight-all.js'
+
+import { useGettext } from 'vue3-gettext'
+import { useThemeStore } from '../composables'
+import { AppConfigObject } from '../apps'
+
+export default defineComponent({
+  name: 'Editor',
+  props: {
+    applicationConfig: { type: Object as PropType<AppConfigObject>, required: true },
+    currentContent: {
+      type: String,
+      required: true
+    },
+    isReadOnly: { type: Boolean, required: false },
+    resource: { type: Object as PropType<Resource>, required: true }
+  },
+  emits: ['update:currentContent'],
+  setup(props, { emit }) {
+    const { current } = useGettext()
+    const themeStore = useThemeStore()
+    const toastuiEditorRef = ref()
+    const toastuiEditor = ref()
+    const config = computed(() => {
+      // TODO: Remove typecasting once vue-tsc has figured it out
+      const { showPreviewOnlyMd = true } = props.applicationConfig as AppConfigObject
+      return { showPreviewOnlyMd }
+    })
+
+    const isMarkdown = computed(() => {
+      return (
+        ['md', 'markdown'].includes(props.resource?.extension) || !unref(config).showPreviewOnlyMd
+      )
+    })
+
+    onMounted(() => {
+      let config: EditorOptions = {
+        el: toastuiEditorRef.value,
+        usageStatistics: false, // sends hostname to google analytics DISABLE
+        initialValue: props.currentContent,
+        hideModeSwitch: true,
+        language: current,
+        height: '100%',
+        plugins: [codeSyntaxHighlight],
+        viewer: props.isReadOnly,
+        events: {
+          change: () => {
+            emit('update:currentContent', toastuiEditor.value.getMarkdown())
+          }
+        },
+        ...(themeStore.currentTheme.isDark && { theme: 'dark' })
+      }
+
+      if (!unref(isMarkdown)) {
+        config = {
+          ...config,
+          toolbarItems: [],
+          initialEditType: 'wysiwyg'
+        }
+      }
+
+      toastuiEditor.value = Editor.factory(config)
+    })
+
```

Comment:
afaict you don't handle updates to props.currentContent, you probably need a watcher on that

Reply:
Added